.changeset/friendly-carrots-remain.md

```diff
@@ -0,0 +1,7 @@
+---
+"@justeattakeaway/pie-icon-button": minor
+"@justeattakeaway/pie-button": minor
+"@justeattakeaway/pie-modal": minor
+---
+
+[Changed] - Use new decorator factories
```

Comment:
Interesting, I think we should have a chat about this as a team to make sure we all know what constitutes a version change. I was under the impression that every PR required a changeset.

Reply:
The way I see it with version bumps, we should always version on code change, mainly so that if we do break something, it's easy to roll back.

The only things that don't require a version bump would be:
- Test only changes (so adding tests, refactoring tests – but no underlying component change)
- Superficial config updates – so any config changes that only affect our repo management, and don't affect the component itself. So, updating an `editorconfig` could be an example of that for instance.

In general, anything that could unintentionally introduce a bug in the component should be versioned, so refactoring should definitely be versioned.

Reply:
Yeah that all sounds reasonable to me!